puppet/modules/logfile/manifests/init.pp

```diff
@@ -0,0 +1,26 @@
+#
+# make syslog log to a particular file for a particular process.
+#
+
+define logfile($process=$title) {
+  $logfile = "/var/log/leap/${title}.log"
+
+  rsyslog::snippet { "50-${name}":
+    content => 'if $programname startswith "${process}" then "${logfile}"
+&~'
```

Comment:
Yeah the content line needs to be in double quotes so the variables will be expanded, but the $programname needs to be escaped so it wont be.